### PR TITLE
Issue 820 grovels

### LINKED ISSUE #820: GRO file with incomplete velocities causes read error.
### Expected behaviour
If a GRO is loaded with the unknown velocities set to zero -- to mimic the GROMACS behaviour.
If a GRO is loaded with no velocities at all, they should not be included.

### Actual behaviour
If no velocities are present in the GRO it is loaded without the velocities attribute.
If velocities are present in the GRO it is loaded with the velocities, but only if they all have velocities set.

### Code to reproduce the behaviour
incomplete_velocities.gro
```
Incomplete velocities
4
 1248DOPC   NC3    1   7.956  12.408   4.949   7.956  12.408   4.949  
 1248DOPC   PO4    2   7.736  12.290   4.671   7.956  12.408   4.949
 1248DOPC   GL1    3   7.760  12.150   4.329
 1248DOPC   GL2    4   7.896  11.896   4.235   7.956  12.408   4.949
  22.28307  22.28307  23.34569
```

``` python
import MDAnalysis as mda

u = mda.Universe('incomplete_velocities.gro')
#crash#

```

### Currently version of MDAnalysis:
0.14.0

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #820

Changes made in this Pull Request:
 - Will now read partially populated GRO velocities

I've had to wrap reading any velocity in a try/except block, which for the adk_oplsaa (42k atoms?) system changes the Universe load time from ~390ms to ~490ms which isn't the best, but I can't think of another way to fix this issue?

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/coordinates/test_gro.py [hunk 1]
```diff
@@ -159,6 +163,25 @@ def test_volume(self):
...
 
+class TestGROIncompleteVels(object):
+    def setUp(self):
+        self.u = mda.Universe(GRO_incomplete_vels)
+
+    def tearDown(self):
+        del self.u
+
+    def test_load(self):
+        assert_(len(self.u.atoms) == 4)
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/coordinates/test_gro.py]: use `assert_equal(len(self.u.atoms), 4)` --- saves the `assert_` import and is clearer.

File: testsuite/MDAnalysisTests/data/grovels.gro [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Incomplete velocities
+4
+ 1248DOPC   NC3    1   7.956  12.408   4.949   7.956  12.408   4.949  
+ 1248DOPC   PO4    2   7.736  12.290   4.671   7.956  12.408   4.949
+ 1248DOPC   GL1    3   7.760  12.150   4.329
+ 1248DOPC   GL2    4   7.896  11.896   4.235   7.956  12.408   4.949
+  22.28307  22.28307  23.34569
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/data/grovels.gro]: maybe add trailing newline?

**Comment by orbeckst** [comment]
First off, I consider it pretty dumb that GRO would allow you to only partially set velocities but if that's what it does then we should support it.

Can't think of a smarter and easier way than try/except. Using `if` on a per line basis is even uglier and probably not faster. If we could read all velocities and then clean up the array, then that would be better, but not sure how to do it, unless we first make an array of strings and then convert... somehow.

By the way, what's in the velocity array in the positions that were not read? Zeros?

**Comment by richardjgowers** [comment]
Ok I've added a warning about partial velocities, anything not set is zero.  Should be good to go now

File: testsuite/MDAnalysisTests/coordinates/test_gro.py [hunk 3]
```diff
@@ -5,11 +5,15 @@
 
 from nose.plugins.attrib import attr
 from numpy.testing import (assert_equal, assert_almost_equal, dec,
-                           assert_array_almost_equal, assert_raises)
+                           assert_array_almost_equal, assert_raises,
+                           assert_)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_gro.py]: `assert_` is not used

File: package/MDAnalysis/coordinates/GRO.py [hunk 4]
```diff
@@ -144,13 +147,19 @@ def _read_first_frame(self):
...
-                for i in range(3):
-                    ts._pos[pos, i] = float(line[20 + cs*i: 20 + cs*(i+1)])
 
-                if not has_velocities:
-                    continue
-                for i, j in enumerate(range(3, 6)):
-                    ts._velocities[pos, i] = float(line[20+cs*j:20+cs*(j+1)])
+                ts._pos[pos] = [line[20 + cs*i:20 + cs*(i+1)] for i in range(3)]
+                try:
+                    velocities[pos] = [line[20 + cs*i:20 + cs*(i+1)] for i in range(3, 6)]
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/GRO.py]: did you import `from six.moves import range` at the beginning of the file?

Comment (kain88-de) [inline package/MDAnalysis/coordinates/GRO.py]: yes it is so you can ignore this.

**Comment by kain88-de** [comment]
If you want to be super pedantic you can check that the warning message is actually correctly printed. I have done that in the xdr tests.

Otherwise it looks good to me.

File: testsuite/MDAnalysisTests/coordinates/test_gro.py (hunk 3 above)
Comment (kain88-de) [inline testsuite/MDAnalysisTests/coordinates/test_gro.py]: This should be removed before I merge it.

**Comment by richardjgowers** [comment]
ok good to go hopefully

**Pull request merged**